hoot-core-test/src/test/cpp/hoot/core/util/HttpTestServer.cpp

```diff
@@ -123,11 +124,9 @@ std::string HttpTestServer::read_request_headers(HttpConnection::HttpConnectionP
 
 void HttpTestServer::write_response(HttpConnection::HttpConnectionPtr& connection, const std::string& response)
 {
-  //  Write the response to the socket asynchronously
-  boost::asio::async_write(connection->socket(), boost::asio::buffer(response),
-                           boost::bind(&HttpConnection::handle_write, connection,
-                                       boost::asio::placeholders::error,
-                                       boost::asio::placeholders::bytes_transferred));
+  //  Write the response to the socket synchronously
+  boost::asio::write(connection->socket(), boost::asio::buffer(response));
+  connection->socket().close();
```

Comment:
Synchronously writing the response here makes the test server a touch less responsive but a lot more stable.

Reply:
Stability is good.